tests/integration/backward_compatible/serialization/compact_compatibility/compact_compatibility_test.py

```diff
@@ -0,0 +1,1691 @@
+
+    def test_contains(self):
+        self.assertFalse(self.queue.contains(OUTER_COMPACT_INSTANCE))
+        self.queue.add(OUTER_COMPACT_INSTANCE)
+        self.assertTrue(self.queue.contains(OUTER_COMPACT_INSTANCE))
+
+    def test_contains_all(self):
+        items = [INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE]
+        self.assertFalse(self.queue.contains_all(items))
+        self.queue.add_all(items)
+        self.assertTrue(self.queue.contains_all(items))
+
+    def test_drain_to(self):
+        target = []
+        self._add_from_another_client(INNER_COMPACT_INSTANCE)
+        self._add_from_another_client(OUTER_COMPACT_INSTANCE)
+        self.assertEqual(2, self.queue.drain_to(target))
+        self.assertEqual([INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE], target)
+
+    def test_offer(self):
+        self.assertTrue(self.queue.offer(OUTER_COMPACT_INSTANCE))
+
+    def test_peek(self):
+        self._add_from_another_client(OUTER_COMPACT_INSTANCE)
+        self.assertEqual(OUTER_COMPACT_INSTANCE, self.queue.peek())
+
+    def test_poll(self):
+        self._add_from_another_client(OUTER_COMPACT_INSTANCE)
+        self.assertEqual(OUTER_COMPACT_INSTANCE, self.queue.poll())
+
+    def test_put(self):
+        self.queue.put(OUTER_COMPACT_INSTANCE)
+        self.assertEqual(OUTER_COMPACT_INSTANCE, self.queue.poll())
+
+    def test_remove(self):
+        self.assertFalse(self.queue.remove(OUTER_COMPACT_INSTANCE))
+        self.queue.add(OUTER_COMPACT_INSTANCE)
+        self.assertTrue(self.queue.remove(OUTER_COMPACT_INSTANCE))
+
+    def test_remove_all(self):
+        items = [INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE]
+        self.assertFalse(self.queue.remove_all(items))
+        self.queue.add_all(items)
+        self.assertTrue(self.queue.remove_all(items))
+
+    def test_retain_all(self):
+        items = [INNER_COMPACT_INSTANCE, OUTER_COMPACT_INSTANCE]
+        self.assertFalse(self.queue.retain_all(items))
+        self.queue.add_all(items + ["x"])
+        self.assertTrue(self.queue.retain_all(items))
+
+    def test_take(self):
+        self._add_from_another_client(OUTER_COMPACT_INSTANCE)
+        self.assertEqual(OUTER_COMPACT_INSTANCE, self.queue.take())
+
+    def _add_from_another_client(self, item):
+        other_client = self.create_client(self.client_config)
+        other_client_queue = other_client.get_queue(self.queue.name).blocking()
+        other_client_queue.add(item)
+
+
+class ReliableTopicCompactCompatibilityTest(CompactCompatibilityBase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.reliable_topic = self.client.get_reliable_topic(random_string()).blocking()
+
+    def tearDown(self) -> None:
+        self.reliable_topic.destroy()
+        super().tearDown()
+
+    def test_publish(self):
+        messages = []
+
+        def listener(message):
+            messages.append(message)
+
+        self.reliable_topic.add_listener(listener)
```

Comment:
In order to test deserialization of compact object in the TopicMessage, can we add the listener from another client?

Reply:
ReliableTopic uses lazy deserialization internally. It reads messages from a ringbuffer in batches. So, we currently don't fully support what you mentioned, hence there is no test for it

Reply:
Can't we support compact objects? I have seen the Node.js code and we can catch SchemaNotFoundError there and fetch the schema. Because there is a promise there.  https://github.com/hazelcast/hazelcast-nodejs-client/blob/061c096a88568d8b99e798259376ce673e5a8b03/src/proxy/topic/ReliableTopicListenerRunner.ts#L65

It may be similar in python

Reply:
I went over the code and I believe although it is possible to do that, I believe it complicates the code a bit and I feel like we are solving the problem at the wrong place. It should have been fixed with the effort around lazy-deserialization.

So, I would like to skip it. WDYT @yuce @srknzl

Reply:
I also vote for skipping this for now. But we should document what won't work with compact objects with unknown schemas (places we do lazy deserialization)